lib/ast/values.js

```diff
@@ -114,11 +114,11 @@ module.exports.Time = Time;
 class AbsoluteTime extends Time {
     constructor(hour, minute, second) {
         super();
-        assert(typeof hour === 'number');
+        assert(typeof hour === 'number' && Number.isFinite(hour));
         this.hour = hour;
-        assert(typeof minute === 'number');
+        assert(typeof minute === 'number' && Number.isFinite(minute));
         this.minute = minute;
-        assert(typeof second === 'number');
+        assert(typeof second === 'number' && Number.isFinite(second));
         this.second = second;
     }
```

Comment:
Shouldn't this be hour < 24, minute, second < 60?

Reply:
Yeah probably. Hour >= 0 too. I don't remember why I added this check though. I think it was to catch NaNs that show up as a result of math with `undefined`.